### PR TITLE
WAR-1253 Estimated run time using previous runs

### PR DESCRIPTION
**Pull request by sugan19**
The filepath is fetched and checks the file is a project/suite/case. 
If it is a case, it fetches the Execution folder path and searches the junit file for the specific case. If there are n junit file available for the specific case then fetches the duration time of all and takes the average of it. 

The estimated time is printed if the previous runs are available.

### CODE REVIEW

File: warrior/Warrior [hunk 1]
```diff
@@ -48,6 +49,77 @@ def update_jira_by_id(jiraproj, jiraid, exec_dir, status):
...
+                    fd = open (filename, 'r')
+                    data = fd.read()
+                    a = data.split('time=')[1].split('\"')[1]
+                    count += 1
+                    total += float(a)
+
+    if count is not 0:
+        avg = total/count
+        time_format = Utils.datetime_utils.get_hms_for_seconds(avg)
+        print "The Estimated Runtime is : ", time_format
```
Comment (bjrr) [inline warrior/Warrior]: use print_utils not python print

File: warrior/Warrior [hunk 2]
```diff
@@ -48,6 +49,77 @@ def update_jira_by_id(jiraproj, jiraid, exec_dir, status):
     else:
         print_info("jiraid not provided, will not update jira issue")
 
+def avg_time_calculate(abs_filepath, default_repo):
+    """ To print the Estimated Runtime for the Project/Suite/Testcase is
+    calculated using the previous junit file runtime values """
+
+    fname = abs_filepath.split('/')[-1].split('.')[0]
```
Comment (bjrr) [inline warrior/Warrior]: / or \ should not be used as they vary from operating system to another. use os.sep , this is a global comment for 
all your code

File: warrior/Warrior [hunk 3]
```diff
@@ -48,6 +49,77 @@ def update_jira_by_id(jiraproj, jiraid, exec_dir, status):
     else:
         print_info("jiraid not provided, will not update jira issue")
 
+def avg_time_calculate(abs_filepath, default_repo):
+    """ To print the Estimated Runtime for the Project/Suite/Testcase is
```
Comment (bjrr) [inline warrior/Warrior]: follow proper docstring structure used in Warrior.

File: warrior/Warrior [hunk 4]
```diff
@@ -48,6 +49,77 @@ def update_jira_by_id(jiraproj, jiraid, exec_dir, status):
...
+    calculated using the previous junit file runtime values """
+
+    fname = abs_filepath.split('/')[-1].split('.')[0]
+    junit_fname = fname + "_junit.xml"
+    file_type = Utils.xml_Utils.getRoot(abs_filepath).tag
+    total = count = 0
+
+    if not default_repo.has_key('ow_resultdir'):
```
Comment (bjrr) [inline warrior/Warrior]: Too many if conditions with no comments. Add comment to each if to explain your logic

File: warrior/Warrior (hunk 2 above)
Comment (venkat1077) [reply warrior/Warrior]: @sugan19  use os.path.basename(abs_filepath).split('.')[0] to avoid using file separators.

File: warrior/Warrior [hunk 5]
```diff
@@ -48,6 +49,77 @@ def update_jira_by_id(jiraproj, jiraid, exec_dir, status):
...
+        if file_type == "Project" or file_type == "TestSuite":
+            junit_file = exec_filepath + "/**/" + junit_fname
+            if os.path.exists(exec_filepath):
+                for filename in glob.iglob(junit_file):
+                    fd = open (filename, 'r')
+                    data = fd.read()
+                    a = data.split('time=')[1].split('\"')[1]
+                    print type(a)
+                    count += 1
+                    total += float(a)
```
Comment (venkat1077) [inline warrior/Warrior]: lines 77-84 are repeating four times, this can be grouped as an internal method.

File: warrior/Warrior [hunk 6]
```diff
@@ -48,6 +49,77 @@ def update_jira_by_id(jiraproj, jiraid, exec_dir, status):
...
+
+        else:
+            file_path = abs_filepath.split(fname)[0]
+            exec_filepath = file_path + "Execution"
+
+        if file_type == "Project" or file_type == "TestSuite":
+            junit_file = exec_filepath + "/**/" + junit_fname
+            if os.path.exists(exec_filepath):
+                for filename in glob.iglob(junit_file):
+                    fd = open (filename, 'r')
```
Comment (venkat1077) [inline warrior/Warrior]: Junit is an XML file, use xml_util methods to parse it rather than treating it as normal text file.

**Comment by sugan19** [comment]
@bjrr @venkat1077 @navvrnd 
Hi Reviewers, 

Few changes are required in the PR, will let you know once it is done.

**Pull request closed without merging**